### PR TITLE
Update release artifact build systems

### LINKED ISSUE #21: Update release artifacts
Update the release artifact builds to new build systems
Ubuntu 22.04
macos 12+
MSVC 2022

### PR DESCRIPTION
**Pull request by nightlark**
Updates release asset build systems to Windows Server (MSVC) 2022, Ubuntu 22.04, and macOS 13 (due to upcoming deprecation of macOS 12 image in December).

Resolves #21

### CODE REVIEW

**Comment by nightlark** [comment]
It looks like quite a few of the matlab versions used by the GitHub Action might not support the Ubuntu 22.04 runners.

**Comment by afisher1** [comment]
@nightlark and @phlptp  there was a conflict with the main branch. can you take a look at my attempted resolution?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -25,14 +25,19 @@ jobs:
         uses: gmlc-tdc/helics-action/install@main
       - name: Set up MATLAB (macOS)
         if: runner.os == 'macOS'
+        uses: matlab-actions/setup-matlab@v2.2.0
```
Comment (phlptp) [inline .github/workflows/ci.yml:28]: this should probably be 2.3 to match the others

Comment (nightlark) [inline .github/workflows/ci.yml:28]:
```suggestion
        uses: matlab-actions/setup-matlab@v2.3.0
```

**Review by afisher1 — APPROVED** [review]

**Pull request merged**